File: ATP/atp.py

class wff:
	def __init__(self, lhs, rhs, symbol):
		self.lhs = lhs
		self.rhs = rhs
		self.symbol = symbol

	def read(formula):
		if formula[0]=='(':
			#find matching bracket
			c = 1
			for i in enumerate(formula[1:]):
				if formula[i]=='(':
					c = c+1
				elif formula[i]==')':
					c = c-1
				if c==0:
					self.lhs =
		else:
			self.lhs = wff(None, None, formula[0])

File: ATP/atp.cpp
#include <iostream>
#include <string>

using namespace std;

class WFF
{
public:
	WFF * lhs;
	WFF * rhs;
	char Symbol;

	WFF(string r)
	{
		read(r);
	}

	WFF(char a)
	{
		init(a);
	}

	void init(a)
	{
		lhs = NULL;
		rhs = NULL;
		Symbol = a;
	}

	void read(string formula)
	{
		if(formula[0]=='(')
		{

		}
		else
		{
			// lhs = WFF(formula[0]);
			WFF * l = new WFF(formula[0]);
			lhs = l;
		}
	}
};